c/compare.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python3
+
+# This script iterates over all tasks and performs the following check:
+# For each file with existing ".c"- and ".i"-file
+# we generate the goto-cc program (i.e., intermediate representation of cbmc)
+# and compare the goto-cc programs with goto-diff.
+# Finding a difference implies a wrong preprocessing.
+
+import glob
+import yaml
+import os
+import shutil
+import subprocess
+from optparse import OptionParser
+
+
+def expand(s):
+    return sorted(glob.glob(s))
+
+
+def getArchitecture(cfgfile):
+    with open(cfgfile, "r") as fp:
+        for line in fp:
+            if line.startswith("Architecture"):
+                return line.split()[1]
+    print("Invalid configuration file", cfgFile)
+    exit(1)
+
+
+def getTasksFromSet(setFile):
+    with open(setFile, "r") as fp:
+        for line in fp:
+            if not line.startswith("#"):
+                for task in expand(line.strip()):
+                    yield task
+
+
+BLACKLIST = set()
+for f in ["floats-esbmc-regression/trunc_nondet_2.i", "*pthread*/*"]:
+    BLACKLIST.update(expand(f))
+
+# parse comand line options and set default values
+parser = OptionParser()
+parser.add_option("-k", "--keep-going", dest="KEEP_GOING", default=False, action="store_true")
+parser.add_option("-v", "--diff", action="store_true", default=False, dest="SHOW_DIFF")
+parser.add_option("--skip-large", action="store_true", default=False, dest="SKIP_LARGE")
+(options, args) = parser.parse_args()
+
+# if the user did not directly specify sets, we select all sets
+SETS=[]
+for arg in (args if args else ["*.set"]):
+    SETS.extend(expand(arg))
+
+# build goto-cc and goto-diff if not available, and then set PATH to find it
+if not shutil.which("goto-cc") or not shutil.which("goto-diff"):
+  if not os.path.exists("../cbmc.git/src/goto-cc/goto-cc"):
```

Comment:
The path `../cbmc.git/` should be extracted into a constant everywhere it appears.